File: leetcode/dia 048/countPrimes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 2:
            return 0
        from math import sqrt
        sieve = [True if i % 2 == 1 else False for i in range(n)]
        sieve[2] == True
        for i in range(3, int(sqrt(n)) + 1, 2):
            if sieve[i]:
                if i * i <= n:
                    up = 3 * i
                    while up < n:
                        sieve[up] = False
                        up += 2 * i
        total = 1
        for i in range(2, n):

            if sieve[i]:

                total += 1
        return total

File: leetcode/dia 049/countPrimes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n <= 2) return 0;
        vector<bool> primes(n, true);
        for (int i = 4; i < n; i += 2)
            primes[i] = false;
        int total = 1;
        for (int i = 3; i < n; i+= 2){
            if (primes[i]){
                total += 1;
                if (i <= sqrt(n)){
                    for (int j = i; j < n; j += 2*i)
                        primes[j] = false;
                }
            }
        }
        return total;
    }
};